apps/src/templates/ContainedLevel.jsx

```diff
@@ -26,9 +32,16 @@ const ContainedLevel = React.createClass({
     } else {
       // No answers yet, disable Run button until there is an answer
       $('#runButton').prop('disabled', true);
+      this.props.setAwaitingContainedResponse(true);
 
       codeStudioLevels.registerAnswerChangedFn(() => {
+        // Ideally, runButton would be declaratively disabled or not based on redux
+        // store state. We might be close to a point where we can do that, but
+        // because runButton is also mutated outside of React (here and elsewhere)
+        // we need to worry about cases where the DOM gets out of sync with the
+        // React layer
```

Comment:
I actually initially did the work to have the runButton be truly connected (not just inside a ProtectedDiv), and things _seemed_ to work, but I decide that this was a bigger/busier PR than justified to be in this PR.